src/org/etools/j1939_84/controllers/Controller.java

```diff
@@ -367,7 +367,9 @@ protected void setupProgress(int maxSteps) {
 
     private void setupRun(ResultsListener listener, J1939 j1939, ReportFileModule reportFileModule) {
         setJ1939(j1939);
-        if (reportFileModule != null) {
+        if (listener instanceof CompositeResultsListener) {
+            compositeListener = (CompositeResultsListener) listener;
+        } else if (reportFileModule != null) {
             compositeListener = new CompositeResultsListener(listener, reportFileModule, partResultRepository);
         } else {
             compositeListener = new CompositeResultsListener(listener, partResultRepository);
```

Comment:
Is this line only used for tests?

Reply:
That was the one used by the PartController and StepController.  It's unused now so I removed it.